File: utilities/validation/python/basic_utils.py
def is_non_empty(value):
    return isinstance(value, str) and bool(value.strip())


def is_email(value):
    if not isinstance(value, str):
        return False
    parts = value.strip().split("@")
    return len(parts) == 2 and bool(parts[0]) and "." in parts[1] and not parts[1].startswith(".") and not parts[1].endswith(".")


def is_within_range(value, minimum, maximum):
    return isinstance(value, (int, float)) and not isinstance(value, bool) and minimum <= value <= maximum

File: utilities/validation/javascript/basic_utils.js
export const isNonEmpty = (value) => typeof value === 'string' && value.trim() !== '';

export const isEmail = (value) => typeof value === 'string' && /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(value);

export const isWithinRange = (value, min, max) => Number.isFinite(value) && value >= min && value <= max;
